cli/azd/pkg/apphost/manifest.go

```diff
@@ -142,8 +150,33 @@ func ManifestFromAppHost(ctx context.Context, appHostProject string, dotnetCli d
 		manifestDir = filepath.Dir(appHostProject)
 	}
 
-	for _, res := range manifest.Resources {
+	manifest.BicepFiles = memfs.New()
+
+	for resourceName, res := range manifest.Resources {
 		if res.Path != nil {
+			if res.Type == "azure.bicep.v0" {
+				e := manifest.BicepFiles.MkdirAll(resourceName, osutil.PermissionDirectory)
+				if e != nil {
+					return nil, e
+				}
+				// try reading as a generated bicep adding the tem-manifest dir
+				content, e := os.ReadFile(filepath.Join(manifestDir, *res.Path))
```

Comment:
Feels like we could simply some of these cases if we moved the logic to after L182, and we know res.Path is absolute?

Reply:
The path for `bicep.v0` is always relative.
When dotnet publishes the manifest, if the module is:
- generated by the appHost, the path is always sibling to the manifest. So we need to use the merge the temp/path were we have published the manifest.
- referenced in appHost and external. The path in this case is referencing a path in the disk where the actual bicep file is. dotnet does not make a copy of that file and put it next to the generated files. So for that case, we need to read the file w/o using the tmp/folder

Reply:
I guess the thing I am still struggling with is the following: On L180-L189 we are concerned with taking paths which are relative to the manifest and turning them into full paths (by appending the app host directory path).  We do this so the rest of the system can always just see full paths if it needs them and not worry about having to know where the app host is so it can resolve the full paths.

The code you are adding here happens before that logic - so you need to concern yourself with adding the app host path and then also dealing with the case where that was wrong (and I suspect using `e != nil` here is not the right test, if there was so IO error but the file existed, we don't want to do go down the "well it didn't work as a relative path, let's try absolute" code-path.  I am wondering if we can move it after L190 so that it only ever sees fully qualified paths to bicep files.  This would let you remove your `// second try reading as relative` case.

Reply:
The path on the bicep becomes automatically decoupled from the appHost's path when the manifest is generated.
If the path on AppHost is something like `/foo/bar.bicep`, the path that is written to the manifest is `<realtive-path-from-manifest-output-to-appHost>/foo/bar.bicep`.
When we use `/tmp/azd-temp42342` as the output, then I would see something like `../../home/user/workspace/app/apphost/foo/bar.bicep`

Does it happen the same with other resources that uses `path` ?

Reply:
For `project` and `dockerfile.v0`, it does work the same. The path is always relative to where the manifest is published. Hence why we do `filepath.Join(manifestDir, *res.Path)` on line 184 and 190. However, we also safeguard against future changes to handle if the path is ever made absolute. We don't need to do `os.Stat`, just simply checking if the path is absolute (string-comparison) is good enough.

Reply:
There might be a bug for project and dockerfile.

The issue I experienced and made me wrote this implementation is that `os.ReadFile()` was unable to read a path like `/tmp/azd-temp393933/../../home/user/project-path` and was returning error.

The difference with project and dockerfile is that, for bicep.v0, we need read the content of the file and not only re-write the path in the manifest.